modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectValidationRuleLocalServiceImpl.java

```diff
@@ -485,6 +517,136 @@ private List<ObjectValidationRule> _getObjectValidationRules(
+				ObjectValidationRuleSettingConstants.
+					MINIMUM_OBJECT_FIELDS_COMPOSITE_KEY)) {
+
+			throw new RequiredObjectFieldException.
+				MustHaveTwoObjectFieldsAtMinimum();
+		}
+
+		if (StringUtil.equals(
+				objectValidationRule.getEngine(),
+				ObjectValidationRuleConstants.ENGINE_TYPE_COMPOSITE_KEY) &&
+			(keyObjectValidationRuleSettings.size() >
+				ObjectValidationRuleSettingConstants.
+					MAXIMUM_OBJECT_FIELDS_COMPOSITE_KEY)) {
+
+			throw new RequiredObjectFieldException.
+				MustHaveFiveObjectFieldsAtMaximum();
+		}
+
+		_deleteObjectValidationRuleSettings(
+			objectValidationRule.getObjectValidationRuleId(),
+			ObjectValidationRuleSettingConstants.
+				NAME_COMPOSITE_KEY_OBJECT_FIELD_ID,
+			keyObjectValidationRuleSettings,
+			_objectValidationRuleSettingPersistence.findByOVRI_N(
+				objectValidationRule.getObjectValidationRuleId(),
+				ObjectValidationRuleSettingConstants.
+					NAME_COMPOSITE_KEY_OBJECT_FIELD_ID));
+		_deleteObjectValidationRuleSettings(
+			objectValidationRule.getObjectValidationRuleId(),
+			ObjectValidationRuleSettingConstants.NAME_OUTPUT_OBJECT_FIELD_ID,
+			outputObjectValidationRuleSettings,
+			_objectValidationRuleSettingPersistence.findByOVRI_N(
+				objectValidationRule.getObjectValidationRuleId(),
+				ObjectValidationRuleSettingConstants.
+					NAME_OUTPUT_OBJECT_FIELD_ID));
+
+		List<ObjectValidationRuleSetting> updatedObjectValidationRuleSettings =
+			new ArrayList<>();
+
+		for (ObjectValidationRuleSetting objectValidationRuleSetting :
+				objectValidationRuleSettings) {
+
+			ObjectValidationRuleSetting
+				serviceBuilderObjectValidationRuleSetting =
+					_objectValidationRuleSettingPersistence.fetchByOVRI_N_V(
+						objectValidationRule.getObjectValidationRuleId(),
+						objectValidationRuleSetting.getName(),
+						objectValidationRuleSetting.getValue());
+
+			if (serviceBuilderObjectValidationRuleSetting != null) {
+				continue;
+			}
+
+			ObjectDefinition objectDefinition =
+				_objectDefinitionPersistence.fetchByPrimaryKey(
+					objectValidationRule.getObjectDefinitionId());
+
+			if (objectDefinition.isApproved()) {
+				ObjectFieldLocalService objectFieldLocalService =
+					_objectFieldLocalServiceSnapshot.get();
+
+				ObjectField objectField =
+					objectFieldLocalService.getObjectField(
+						GetterUtil.getLong(
+							objectValidationRuleSetting.getValue()));
+
+				Table<?> table = objectFieldLocalService.getTable(
+					objectValidationRule.getObjectDefinitionId(),
+					objectField.getName());
+
+				Column<?, ?> column = table.getColumn(
+					objectField.getDBColumnName());
+
+				ObjectEntryLocalService objectEntryLocalService =
+					_objectEntryLocalServiceSnapshot.get();
+
+				long objectEntriesCount =
+					objectEntryLocalService.getObjectEntriesCount(
+						0, objectDefinition, column.isNotNull());
+
+				if (objectEntriesCount > 0) {
+					throw new RequiredObjectFieldException.
+						MustNotUseFilledObjectField();
+				}
```

Comment:
@mateussgn do we need to make this validation even for the setting `NAME_OUTPUT_OBJECT_FIELD_ID`?

Reply:
Does not need.

Reply:
No need for `NAME_OUTPUT_OBJECT_FIELD_ID`.